mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -620,29 +622,24 @@ private[ml] object RandomForest extends Logging {
     }
   }
 
-  /**
-   * Calculate the information gain for a given (feature, split) based upon left/right aggregates.
-   * @param leftImpurityCalculator left node aggregates for this (feature, split)
-   * @param rightImpurityCalculator right node aggregate for this (feature, split)
-   * @return information gain and statistics for split
-   */
-  private def calculateGainForSplit(
+  private def calculateImpurityStats(
       leftImpurityCalculator: ImpurityCalculator,
       rightImpurityCalculator: ImpurityCalculator,
-      metadata: DecisionTreeMetadata,
-      impurity: Double): InformationGainStats = {
+      metadata: DecisionTreeMetadata): ImpurityStats = {
     val leftCount = leftImpurityCalculator.count
     val rightCount = rightImpurityCalculator.count
 
+    val totalCount = leftCount + rightCount
+    val parentImpurityCalculator = leftImpurityCalculator.copy.add(rightImpurityCalculator)
+    val impurity = parentImpurityCalculator.calculate()
```

Comment:
This impurity is already available, so why not pass it in as an argument to avoid recalculating it?